derive/src/error.rs

```diff
@@ -0,0 +1,100 @@
+use std;
+
+use json::JsonError;
+
+/// The result type for this procedural macro.
+pub(crate) type Result<T> = std::result::Result<T, Error>;
+
+/// Represents errors that may be encountered in
+/// invokations of this procedural macro.
+#[derive(Debug)]
+pub(crate) struct Error {
+    /// The kind of this error.
+    kind: ErrorKind,
+}
+
+/// Kinds of errors that may be encountered in invokations
+/// of this procedural macro.
+#[derive(Debug)]
+pub(crate) enum ErrorKind {
+    /// An error that occured upon a JSON operation.
+    JsonError(JsonError),
+    /// When there was an invalid number of arguments passed to `eth_abi`.
+    InvalidNumberOfArguments {
+        /// The number of found arguments.
+        found: usize,
```

Comment:
why not add actual (expected) number of arguments as a variant field also?

or, if not, make it `InvalidNumberOfArguments(usize)` ?